File: problems/1422_maximum-score-after-splitting-a-string/main.go
// +build 186_0

package main

func maxScore(s string) int {
	var cnt1 int
	for _, r := range s {
		if r == '1' {
			cnt1++
		}
	}

	var result, left0 int
	if s[0] == '0' {
		result = 1 + cnt1
		left0++
	} else {
		result = cnt1 - 1
		cnt1--
	}

	for i := 1; i < len(s)-1; i++ {
		if s[i] == '0' {
			left0++
		} else {
			cnt1--
		}

		result = max(result, left0+cnt1)
	}

	return result
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	println(maxScore("011101"))
	println(maxScore("00111"))
	println(maxScore("1111"))
	println(maxScore("1011011"))
}

File: problems/1422_maximum-score-after-splitting-a-string/__main__.py
class Solution:
    def maxScore(self, s: str) -> int:
        counter = collections.Counter(s)
        ret, cnt0, cnt1 = 0, 0, 0
        for i in range(len(s) - 1):
            if s[i] == "0":
                cnt0 += 1
            else:
                cnt1 += 1
            ret = max(ret, cnt0 + counter["1"] - cnt1)
        return ret
